radixdlt-core/radixdlt/src/integration/java/com/radixdlt/harness/simulation/monitors/consensus/MaxLedgerSyncLagInvariant.java

```diff
@@ -108,9 +108,13 @@ public MaxLedgerSyncLagInvariant(long maxLag) {
                           Observable.just(
                               new TestInvariantError(
                                   String.format(
-                                      "Node %s ledger sync lag exceeded maximum of %s state"
-                                          + " versions",
-                                      e.getKey(), maxLag))))
+                                      "Node %s ledger sync lag (%s) at version %s exceeded maximum"
+                                          + " of %s state versions",
+                                      e.getKey(),
+                                      maxStateVersion
+                                          - e.getValue().get(CounterType.LEDGER_STATE_VERSION),
+                                      e.getValue().get(CounterType.LEDGER_STATE_VERSION),
+                                      maxLag))))
```

Comment:
Extracting lambda body into the dedicated method might help reduce nesting and better separate responsibilities.